File: Basics_with_JavaScript/1112 While Loop/12 While loop Excercise/04. Walking.js
function vladi(array){
    let goal = 10000;
    let walked_steps = 0;
    let steps = 0
    let index = 0;
    let isGoalReached = false;
    let isGoingHome = false;
    let command = array[index];
    index++;

    while (true) {
        if (command === "Going home") {
            isGoingHome = true;
            command = Number(array[index]);
            walked_steps += command;
            if (walked_steps >= goal) {
                isGoalReached = true;
            }
            break;
        }
        
        steps = Number(command);
        walked_steps += steps;
        if (walked_steps >= goal) {
            isGoalReached = true;
            break;
        }
        command = array[index];
        index++;
    }

    let diff = Math.abs(goal - walked_steps)
    if (isGoalReached) {
        console.log(`Goal reached! Good job!`)
        console.log(`${diff} steps over the goal!`);
    } else {
        console.log(`${diff} more steps to reach goal.`);
    }
    }
   
    vladi(["1000",
    "1500",
    "2000",
    "6500"])
    ;
    vladi(["1500",
    "3000",
    "250",
    "1548",
    "2000",
    "Going home",
    "2000"])
    ;
    vladi(["1500",
    "300",
    "2500",
    "3000",
    "Going home",
    "200"])
    ;
    vladi(["125",
    "250",
    "4000",
    "30",
    "2678",
    "4682"])
    ;

File: Basics_with_Python/PythonBasicCourse/5_While Loop/12_While Loop - Exercise/04. Walking.py
goal = 10000
step_counter = 0
steps_walked = input()

while True:
    if steps_walked == "Going home":
        break
    else:
        steps_walked = int(steps_walked)
        step_counter += steps_walked
        if step_counter >= goal:
            print("Goal reached! Good job!")
            print(f"{step_counter - goal} steps over the goal!")
            break
        steps_walked = input()

if steps_walked == "Going home":
    steps_walked = int(input())
    step_counter += steps_walked
    if step_counter < goal:
        print(f"{goal - step_counter} more steps to reach goal.")
    else:
        print("Goal reached! Good job!")
        print(f"{step_counter - goal} steps over the goal!")


#
# while steps_walked != "Going home":
#     steps_walked = int(steps_walked)
#     step_counter += steps_walked
#     if step_counter >= goal:
#         print("Goal reached! Good job!")
#         print(f"{step_counter - goal} steps over the goal!")
#         break
#     steps_walked = input()
#
# if steps_walked == "Going home":
#     steps_walked = int(input())
#     step_counter += steps_walked
#     if step_counter < goal:
#         print(f"{goal - step_counter} more steps to reach goal.")
#     else:
#         print("Goal reached! Good job!")
#         print(f"{step_counter - goal} steps over the goal!")
